### PR TITLE
feat(java): allow streaming byte requests

### PR DESCRIPTION
**Pull request by dcb6**

### CODE REVIEW

File: seed/java-sdk/bytes/src/main/java/com/seed/bytes/resources/service/ServiceClient.java [hunk 1]
```diff
@@ -58,4 +61,9 @@ public void upload(byte[] request, RequestOptions requestOptions) {
             throw new SeedBytesException("Network error executing HTTP request", e);
         }
     }
+
+    // Overload for backward compatibility
+    public void upload(byte[] request, RequestOptions requestOptions) {
```
Comment (dcb6) [inline seed/java-sdk/bytes/src/main/java/com/seed/bytes/resources/service/ServiceClient.java:66]: thoughts on moving the direct bytes request to this?

Claude gave me these downsides but to me seems worth the cleanliness

> Potential downsides of the upload overload:

Memory Usage:

The original implementation used the byte array directly.
The new overload wraps the byte array in a ByteArrayInputStream, which is then wrapped in a FileStream.
While ByteArrayInputStream is generally efficient, it does create an additional object in memory.


Performance Overhead:

There's a slight performance cost in creating the additional objects (ByteArrayInputStream and FileStream).
The streaming approach might introduce a small amount of overhead compared to sending the byte array directly, especially for small files.

File: seed/java-sdk/bytes/src/main/java/com/seed/bytes/core/FileStream.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+import java.io.InputStream;
+import java.util.Objects;
+import okhttp3.MediaType;
+import okhttp3.RequestBody;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Represents a file stream with associated metadata for file uploads.
+ */
+public class FileStream {
```
Comment (dcb6) [inline seed/java-sdk/bytes/src/main/java/com/seed/bytes/core/FileStream.java:12]: we don't actually need this for bytes requests, going to delete unless anyone votes I keep it around

File: seed/java-sdk/bytes/src/main/java/com/seed/bytes/resources/service/ServiceClient.java (hunk 1 above)
Comment (dsinghvi) [reply seed/java-sdk/bytes/src/main/java/com/seed/bytes/resources/service/ServiceClient.java:69]: great idea

**Review by dsinghvi — APPROVED** [review]

File: seed/java-sdk/bytes/src/main/java/com/seed/bytes/core/FileStream.java (hunk 2 above)
Comment (dsinghvi) [reply seed/java-sdk/bytes/src/main/java/com/seed/bytes/core/FileStream.java]: awesome, may be useful to keep around to make it better for multipart (just as a uncopied utility)

File: generators/java/sdk/versions.yml [hunk 3]
```diff
@@ -1,10 +1,19 @@
...
+  version: 2.3.0
+
 - changelogEntry:
     - summary: |
         Bump Jackson version to latest (2.17.2)
       type: chore
   createdAt: '2024-09-26'
   irVersion: 46
-  version: 1.2.0
+  version: 2.2.0
```
Comment (dcb6) [inline generators/java/sdk/versions.yml:16]: this was a blunder 😅 

thankfully, we never previously released a `1.2.0` and niels won't be using that until next week
I deleted this erroneous `1.2.0` release from docker hub

**Pull request merged**